### PR TITLE
dont remove all adapters if only db migration is done

### PR DESCRIPTION
**Pull request by foxriver76**
else this can take a while on sd card systems and is really annoying ;-)

### CODE REVIEW

**Comment by AlCalzone** [comment]
This could explain my issues when I did this in dev-server.

**Comment by foxriver76** [comment]
> This could explain my issues when I did this in dev-server.

I think it was actually a missing parameter which lead to the case, that the cb function has never been executed. See https://github.com/ioBroker/ioBroker.js-controller/commit/f536c01a0863218d8036b093d4b19211b7eeee7b#diff-6923d54cffa14e01b56d83eb75b128ad8aeaa0a87c1f4e37a9a69da0ddae74b0R557

**Comment by AlCalzone** [comment]
I also got missing adapters after the half-done migration, so I guess both happened to me.

File: packages/controller/lib/setup.js [hunk 1]
```diff
@@ -2725,37 +2729,39 @@ async function delStates() {
...
-            // Clean up states
-            const keysCount = await delStates();
-
-            if (callback) {
-                return void callback(keysCount);
-            }
-        });
+        await objects.destroyDBAsync();
+        // Clean up states
+        return delStates();
```
Comment (Apollon77) [inline packages/controller/lib/setup.js:2742]: return promise here, but below return count only??

File: packages/controller/lib/setup.js [hunk 2]
```diff
@@ -1121,17 +1121,21 @@ async function processCommand(command, args, params, callback) {
...
-                        restartController();
-                        console.log(`Restarting ${tools.appName}...`);
-                        return void callback();
-                    });
+                    } catch {
+                        // ignore
+                    }
+                    restartController();
+                    console.log(`Restarting ${tools.appName}...`);
+                    return void callback();
```
Comment (AlCalzone) [inline packages/controller/lib/setup.js:1138]:
```suggestion
                    callback();
```
return void at the end of a function doesn't make a ton of sense.

File: packages/controller/lib/setup.js (hunk 1 above)
Comment (AlCalzone) [reply packages/controller/lib/setup.js:2742]: This works (async function!), but isn't consistent, and the code below is more obvious IMO.

File: packages/controller/lib/setup.js [hunk 3]
```diff
@@ -2725,37 +2729,39 @@ async function delStates() {
...
+        try {
+            const res = await objects.getObjectListAsync({ startkey: '\u0000', endkey: '\u9999' });
+            if (res.rows.length) {
                 console.log(`clean ${res.rows.length} objects...`);
                 ids = res.rows.map(e => e.id);
             }
-            await delObjects(ids);
-            // Clean up states
-            const keysCount = await delStates();
+        } catch {
```
Comment (AlCalzone) [inline packages/controller/lib/setup.js:2753]: Does node 12 support omitting the error param yet? Not sure.

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/controller/lib/setup.js (hunk 2 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:1138]: not touched but yes, can simply just call cb

File: packages/controller/lib/setup.js (hunk 1 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:2742]: yes basically the same as assigning to const and then returining, just an unnecessary step but yes just seeing it right now that its in two steps below and if you find it better readable lets to it like that

File: packages/controller/lib/setup.js (hunk 3 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:2753]: yes, we do this a lot

**Pull request merged**